src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -62,6 +62,11 @@ internal abstract class PENamedTypeSymbol : NamedTypeSymbol
         /// </summary>
         private TypeKind _lazyKind;
 
+        /// <summary>
+        /// Lazily initiated HasEmbeddedAttribute
+        /// </summary>
+        private ThreeState _lazyHasEmbeddedAttribute = ThreeState.Unknown;
```

Comment:
`UncommonProperties` are only instantiated for `PENamedTypeSymbols` with uncommon attributes. The instances of `UncommonProperties` are not shared.